aws/cf_template.go

```diff
@@ -346,3 +351,26 @@ func generateTemplate(spec *stackSpec) (string, error) {
 
 	return string(stackTemplate), nil
 }
+
+func denyInternalTraffic(template *cloudformation.Template, listenerName string) {
+	template.AddResource("HTTPRuleBlockInternalTraffic", &cloudformation.ElasticLoadBalancingV2ListenerRule{
+		Conditions: &cloudformation.ElasticLoadBalancingV2ListenerRuleRuleConditionList{
+			cloudformation.ElasticLoadBalancingV2ListenerRuleRuleCondition{
+				Field:  cloudformation.String("host-header"),
+				Values: cloudformation.StringList(cloudformation.String("*.ingress.cluster.local")),
+			},
+		},
+		Actions: &cloudformation.ElasticLoadBalancingV2ListenerRuleActionList{
+			cloudformation.ElasticLoadBalancingV2ListenerRuleAction{
+				Type: cloudformation.String("fixed-response"),
+				FixedResponseConfig: &cloudformation.ElasticLoadBalancingV2ListenerRuleFixedResponseConfig{
+					ContentType: cloudformation.String("text/plain"),
+					MessageBody: cloudformation.String("Not Found"),
+					StatusCode:  cloudformation.String("404"),
```

Comment:
Not 401?

Reply:
We can discuss that further, but at the moment that's a draft with the code I used for a POC in order to load test. It shouldn't be reviewed.